save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/SettingsView.kt

```diff
@@ -133,11 +142,13 @@ fun saveUser(
     settingsInputFields: SettingsInputFields,
     setFieldsValidation: FieldsStateSetter,
 ) = useDeferredRequest {
+    // this new user info will be sent to backend and also will be set in setter,
+    // so frontend will recalculate it on the fly at least for SettingsView (need to extend it later)
+    val newUserInfo = settingsInputFields.toUserInfo(props.userInfo!!)
     val response = post(
         "$apiUrl/users/save",
         jsonHeaders,
-
-        Json.encodeToString(settingsInputFields.toUserInfo(props.userInfo!!)),
+        Json.encodeToString(newUserInfo),
```

Comment:
I dare you I double dare you to rename this method to `useSaveUser`

![](https://cdn.quotesgram.com/img/3/5/618438753-tumblr_kuugs1ZV8r1qa2ylwo1_500.jpg)

Reply:
ok, you won